### PR TITLE
Update to MongoDB Component Doc

### PR DESCRIPTION
**Pull request by LuceCarter**
## Summary

Added some additional information regarding support for MongoDB Atlas. 
Also removed a section around the configuration extensions which doesn't appear to be supported by the component.

Fixes #Issue_Number (if available)
744

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/database/mongodb-component.md](https://github.com/dotnet/docs-aspire/blob/9b99a738cabf7425a39006ea44bcf38eb33d5efe/docs/database/mongodb-component.md) | [docs/database/mongodb-component](https://review.learn.microsoft.com/en-us/dotnet/aspire/database/mongodb-component?branch=pr-en-us-1410) |

### CODE REVIEW

**Comment by LuceCarter** [comment]
@dotnet-policy-service agree [company="MongoDB"]

**Comment by LuceCarter** [comment]
@dotnet-policy-service agree company="MongoDB"

File: docs/database/mongodb-component.md [hunk 1]
```diff
@@ -87,6 +88,7 @@ var mongodb = mongo.AddDatabase("mongodb");
 var myService = builder.AddProject<Projects.MyService>()
                        .WithReference(mongodb);
 ```
+**N.B.**: If using MongoDB Atlas, you do not need the call to ```.AddDatabase("mongodb");``` as the database creation will automatically be handled by Atlas.
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:91]:
```suggestion

If using MongoDB Atlas, you don't need the call to `.AddDatabase("mongodb");` as the database creation will automatically be handled by Atlas.
```

File: docs/database/mongodb-component.md [hunk 2]
```diff
@@ -102,6 +104,8 @@ builder.AddMongoDBClient("MongoConnection");
 
 And then the connection string will be retrieved from the `ConnectionStrings` configuration section:
 
+MongoDB Example:
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:107]:
```suggestion
Consider the following MongoDB example JSON configuration:
```

File: docs/database/mongodb-component.md [hunk 3]
```diff
@@ -110,29 +114,18 @@ And then the connection string will be retrieved from the `ConnectionStrings` co
...
 ```
 
-For more information on how to format this connection string, see [MongoDB: ConnectionString documentation](https://www.mongodb.com/docs/v3.0/reference/connection-string).
-
-### Use configuration providers
-
-The .NET Aspire MongoDB database component supports <xref:Microsoft.Extensions.Configuration?displayProperty=fullName>. It loads the `MySqlConnectorSettings` from configuration files such as _:::no-loc text="appsettings.json":::_ by using the `Aspire:MongoDB:Driver` key. If you have set up your configurations in the `Aspire:MongoDB:Driver` section, you can just call the method without passing any parameter.
-
-The following example shows an _:::no-loc text="appsettings.json":::_ file that configures some of the available options:
+MongoDB Atlas Eexmple:
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:117]:
```suggestion
Consider the following MongoDB Atlas example JSON configuration:
```

File: docs/database/mongodb-component.md [hunk 4]
```diff
@@ -110,29 +114,18 @@ And then the connection string will be retrieved from the `ConnectionStrings` co
...
-      "Driver": {
-        "ConnectionString": "mongodb://server:port/test",
-        "DisableHealthChecks": false,
-        "HealthCheckTimeout": 10000,
-        "DisableTracing": false
-      },
-    }
+  "ConnectionStrings": {
+    "MongoConnection": "mongodb+srv://username:password@server.mongodb.net/",
   }
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:123]: Please revert these changes, I understand the need to show differences between connection strings for MongoDB and MongoDB with Atlas, but this configuration is still valid and needs to be reflected in docs.

File: docs/database/mongodb-component.md [hunk 5]
```diff
@@ -110,29 +114,18 @@ And then the connection string will be retrieved from the `ConnectionStrings` co
...
 }
 ```
 
+For more information on how to format this connection string, see [MongoDB: ConnectionString documentation](https://www.mongodb.com/docs/v3.0/reference/connection-string).
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:127]: Re-add removed section.

```suggestion
For more information on how to format this connection string, see [MongoDB: ConnectionString documentation](https://www.mongodb.com/docs/v3.0/reference/connection-string).

### Use configuration providers

The .NET Aspire MongoDB database component supports <xref:Microsoft.Extensions.Configuration?displayProperty=fullName>. It loads the `MySqlConnectorSettings` from configuration files such as _:::no-loc text="appsettings.json":::_ by using the `Aspire:MongoDB:Driver` key. If you have set up your configurations in the `Aspire:MongoDB:Driver` section, you can just call the method without passing any parameter.

The following example shows an _:::no-loc text="appsettings.json":::_ file that configures some of the available options:

```json
{
  "Aspire": {
    "MongoDB": {
      "Driver": {
        "ConnectionString": "mongodb://server:port/test",
        "DisableHealthChecks": false,
        "HealthCheckTimeout": 10000,
        "DisableTracing": false
      },
    }
  }
```

File: docs/database/mongodb-component.md [hunk 6]
```diff
@@ -179,7 +172,6 @@ The .NET Aspire MongoDB database component will emit the following Tracing activ
 The .NET Aspire MongoDB database component doesn't currently expose any OpenTelemetry metrics.
 
 ## See also
```
Comment (IEvangelist) [inline docs/database/mongodb-component.md:174]: Re-add space after heading to fix MD lint issue.

```suggestion
## See also

```

**Review by IEvangelist — CHANGES_REQUESTED** [review]

File: docs/database/mongodb-component.md (hunk 5 above)
Comment (LuceCarter) [reply docs/database/mongodb-component.md:128]: Happy to add this back. However, is it still MySqlConnectorSettings when its MongoDB? I don't know the answer to that.
It also mentions that if you use this section, you don't need to pass any parameter, which I didn't find was the case. .AddMongoClient requires a connection name.

Comment (IEvangelist) [reply docs/database/mongodb-component.md:128]: I fixed the suggestion, it did have a typo and had incorrect info. Thanks for catching that.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏
